tests/unit/assessment/test_pipelines.py

```diff
@@ -106,25 +102,50 @@ def test_pipeline_without_owners_should_have_empty_creator_name():
             creator_user_name=None,
             latest_updates=None,
             name="New DLT Pipeline",
-            pipeline_id="0112eae7-9d11-4b40-a2b8-6c83cb3c7407",
+            pipeline_id="pipeline_id_without_conf",
             run_as_user_name="abcde.defgh@databricks.com",
             state=PipelineState.IDLE,
         )
     ]
 
-    ws = Mock()
+    ws = workspace_client_mock(clusters="no-spark-conf.json")
     ws.pipelines.list_pipelines.return_value = sample_pipelines
-    ws.pipelines.get().spec.configuration = {}
+    ws.workspace.export().content = "JXNoCmVjaG8gIj0="
+    ws.dbfs.read().data = "JXNoCmVjaG8gIj0="
     mockbackend = MockBackend()
     PipelinesCrawler(ws, mockbackend, "ucx").snapshot()
     result = mockbackend.rows_written_for("hive_metastore.ucx.pipelines", "append")
 
     assert result == [
         PipelineInfo(
-            pipeline_id="0112eae7-9d11-4b40-a2b8-6c83cb3c7407",
+            pipeline_id="pipeline_id_without_conf",
             pipeline_name="New DLT Pipeline",
             creator_name=None,
             success=1,
             failures="[]",
         )
     ]
+
+
+def test_pipeline_assessment_with_no_cluster_config():
+    sample_pipelines = [
+        PipelineStateInfo(
+            cluster_id=None,
+            creator_user_name="abcde.defgh@databricks.com",
+            latest_updates=None,
+            name="New DLT Pipeline",
+            pipeline_id="pipeline_id_with_conf",
+            run_as_user_name="abcde.defgh@databricks.com",
+            state=PipelineState.IDLE,
+        )
+    ]
+
+    ws = workspace_client_mock(clusters="no-spark-conf.json")
+    ws.workspace.export().content = "JXNoCmVjaG8gIj0="
+    ws.dbfs.read().data = "JXNoCmVjaG8gIj0="
```

Comment:
do we need these for pipeline tests?

Reply:
Yup I added this to cover the negative scenario for full coverage. But with the addition of the spec in json we can remove this. Removed this from test.